### ISSUE TITLE
Convert CAPs output to leaspy-complient input

### ISSUE THREAD

**Issue by omar-rifai** [aramis-lab/clinica#276]
[Leaspy](https://gitlab.com/icm-institute/aramislab/leaspy/) is a software package for statistical analysis of longitudinal medical data. There has been an interest in using Clinics-processed data as input to Leaspy.

As such, it would be helpful to have a integrated converter from Clinica pipelines outputs (CAPS) into Leaspy inputs.

**Comment by Symcies** [comment]
Leaspy consider three type of data : 
- raw  : raw inputs in any possible format
- clean : standardize inputs, i.e. one csv file per type of data where each row is a patient visit, and that contains two columns : 'ID' for the patient ID, and 'TIME' for the patient age at the corresponding visit. The other columns corresponds to the feature values at the corresponding visit
- leaspy-ready : normalized between 0 and 1, potential missing value removal, aggregation, & so on

Currently, Leaspyers have to write a code to parse the CAPS data (considered "raw") to convert them into the clean format. This code has been written multiple times (@JulietteOrtholand  or Etienne Maheux) and should be stored somewhere once and for all.

On top of the dev specs as for the function that would do the "CAPS => Leaspy input" parsing + conversion, further thoughts should be given to where the following code stands. To be more precise, the previous formats are actually part of the proprietary Leaspype code, which is a wrapper around Leaspy.

As for now, this code will remain proprietary. So, should the parser be : 
- In Clinica : (+) part of Clinica converters. (-) Maybe a bit specific
- In Leaspy : (+) Not directly related to Clinica, but may be worth living in the utils. (-) Not related to Clinica
- In Leaspype (+) Where it will serve the most. (-) Proprietary code
- Elsewhere : (+) Could be a submodule of both Clinica & Leaspy/Leaspype. (-) Very small package which will be tighkly related to Leaspy & Clinica

**Comment by Symcies** [comment]
_Another point related to where the code should be_

In fact, the `clean`  data follow some conventions, in terms of naming (e.g. MMSE is `MMSE_TOT`) that are different from Clinica. These conventions are in Leaspype, which makes the code more likely to host the final convertor. [Additional point : We will probably need to modify the conventions on our side to properly host / deal with imaging data].

Also further thoughts should be given to : 
- What type of data do we want to convert : any CAPS data? Some volumetric data? Free-surfer output? SPM? Other? 
- How do we convert non-imaging data, esp. clinical/cognitive assessments? Does Leaspype have a convertor of its own for this data, and uses the Clinica-parser for the imaging? Other ideas?

**Comment by mdiazmel** [comment]
Hi @Symcies , 

Thanks for your comments concerning this issue.

> As for now, this code will remain proprietary. So, should the parser be :

To answer your question about where the parser should live, I will exclude the two first options: 

- If I understand correctly, with the first option (the parser included in Clinica), we can think about a convertor, let's say the `caps-2-leaspy` convertor, that will wrap some proprietary code. To be 100% functional, the proprietary code must be a requirement for Clinica package (and, in my opinion, we should try to avoid this).  
- For the second option (in Leaspy), the link with Clinica will be too weak and it should implies to add Clinica as dependency of Leaspy (in order to use the CAPS tools). Maybe, it isn't the best thing for deploying Leaspy.

Among the two last option, I'll be up for creating a new package, that will depend on Leaspy and Clinica. It implies a little bit more of work but it seems as a proper solution.

**Comment by ghisvail** [comment]
> the link with Clinica will be too weak and it should implies to add Clinica as dependency of Leaspy (in order to use the CAPS tools)

Not necessarily. Weak / optional dependencies are already well supported in Python with [extras_require](https://setuptools.readthedocs.io/en/latest/userguide/dependency_management.html#optional-dependencies). I'd rather do `pip install leaspy[all]` than `pip install leaspy <hang on what's the name of the other caps thing? dammit, gonna have to look in the docs>`.

**Comment by MKVatinyan** [comment]
Hi all,
I have never used Clinica and I don't have any use cases in mind, so it's a bit hard to participate to the discussion for me, sorry. 
- We are talking about converting the output of the anatomical pipelines of Clinica right ? (linear, volume, freesurfer)
-  If i understand correctly those pipelines generate several files, are there some specific files that need to be converted ? (I was shown the example of aseg.stats in freesurfer_cross_sectional output)
- For the parser location maybe both propositions can be merged ? We can have a third package caps2leaspy and add [clinica, caps2leaspy] as optional dependencies in leaspy and maybe [leaspy, caps2leaspy] as optional dependencies in clinica ?

**Comment by ghisvail** [comment]
> We can have a third package caps2leaspy and add [clinica, caps2leaspy] as optional dependencies in leaspy and maybe [leaspy, caps2leaspy] as optional dependencies in clinica ?

I think you mean introducing `caps2leaspy` dependent on `clinica` and `leaspy` listing `caps2leaspy` as optional dependency to both `clinica` and `leaspy` right?

**Comment by MKVatinyan** [comment]
> > We can have a third package caps2leaspy and add [clinica, caps2leaspy] as optional dependencies in leaspy and maybe [leaspy, caps2leaspy] as optional dependencies in clinica ?
> 
> I think you mean introducing `caps2leaspy` dependent on `clinica` and `leaspy`, and listing `caps2leaspy` as optional dependency to both `clinica` and `leaspy` right?

yes
